go/mcap/lexer.go

```diff
@@ -55,13 +55,37 @@ func (e *ErrTruncatedRecord) Unwrap() error {
 	return io.ErrUnexpectedEOF
 }
 
-// ErrBadMagic indicates the lexer has detected invalid magic bytes.
+type magicLocation int
+
+const (
+	magicLocationStart magicLocation = iota
+	magicLocationEnd
+)
+
+func (m magicLocation) String() string {
+	switch m {
+	case magicLocationStart:
+		return "start"
+	case magicLocationEnd:
+		return "end"
+	default:
+		return "unknown"
+	}
+}
+
+// ErrBadMagic indicates invalid magic bytes were detected
 type ErrBadMagic struct {
-	actual []byte
+	location magicLocation
+	actual   []byte
 }
```

Comment:
I considered moving this type into a new file, "errors.go" to collect public errors in that place and maybe foster more straightforward reuse - but held off in the end since it started feeling like a larger change.